inheritance1/Byn.java

```diff
@@ -0,0 +1,95 @@
+public class Byn implements Comparable<Byn>{
+    private int value;
+
+    public Byn(){
+    }
+
+    public Byn(int value){
+        this.value = value;
+    }
+
+    public Byn(int rubs, int coins){
+        this(rubs * 100 + coins);
+    }
+
+    public Byn(Byn byn){
+        this(byn.value);
+    }
+
+    public String convert(){
+        return value / 100 + "." + value / 10 % 10 + value % 10;
+    }
+
+    @Override
+    public String toString(){
+        return String.format("%s", convert());
+    }
+
+    @Override
+    public int compareTo(Byn byn){
+        return value - byn.value;
+    }
+
+    public Byn multiply(int amount){
+        this.value = this.value * amount;
+        return this;
+    }
+
+    public Byn divide(int amount){
+        this.value = this.value / amount;
+        return this;
+    }
+
+    public Byn add(Byn amount){
+        this.value = value + amount.value;
+        return this;
+    }

```

Comment:
Why first one value with "this", but second not? And why you need this anyway(same question for all arithmetic operations)?